### PR TITLE
don't remove meaningful empty elements

### PR DESCRIPTION
**Pull request by alexlamsl**
#536 mentions `<iframe>` - I've further identified `<audio>` and `<video>`.

### CODE REVIEW

File: src/htmlminifier.js [hunk 1]
```diff
@@ -481,18 +481,19 @@
...
-      for (var i = attrs.length - 1; i >= 0; i--) {
-        if (attrs[i].name === 'src') {
-          return false;
+    switch (tag) {
+      case 'audio':
+      case 'iframe':
+      case 'textarea':
+      case 'video':
+        return false;
+      case 'script':
```
Comment (kangax) [inline src/htmlminifier.js:498]: Shouldn't video, audio and iframe act the same way as script? Only be removed when they don't have "src" attr?

Comment (alexlamsl) [inline src/htmlminifier.js:498]: I was being conservative - but yeah let's go with yours.

Can you think of other tags while we are at this?

**Comment by alexlamsl** [comment]
@kangax commit amended

**Comment by kangax** [comment]
Perhaps `<object>`?

**Comment by alexlamsl** [comment]
Assuming `<object data="...">` then?

**Comment by alexlamsl** [comment]
Right, added `<applet code="...">` while we are at this.

**Comment by alexlamsl** [comment]
... I've just discovered `<spacer>` as well :sweat:

**Comment by alexlamsl** [comment]
Okay, looks like only Netscape and very old Mozilla browers support `<spacer>`, so I guess it's not a big deal...?

**Comment by kangax** [comment]
HTML is hard :D

**Comment by alexlamsl** [comment]
I'm actually thinking about `<i class="fa fa-camera-retro">` and same with `<span>` - they are commonly used to put out these font-based icons and `removeEmptyElements` would kill them.

**Comment by kangax** [comment]
Not sure it's worth adding support for spacer. It's deprecated, no?

**Comment by alexlamsl** [comment]
Probably not - it's one of those "no harm, let's throw in the kitchen sink" moment :ghost: 

I'll take it back out then...

**Comment by alexlamsl** [comment]
So what are your thoughts on sparing `<i>` and `<span>` when `class` attribute is defined?

**Comment by kangax** [comment]
> So what are your thoughts on sparing <i> and <span> when class attribute is defined?

That's expected. We should make it very clear in the docs. Ideally, they need to be ignored via config or something.

**Comment by alexlamsl** [comment]
I guess that link to your blog post has enough warning signs on it.

**Comment by alexlamsl** [comment]
`<spacer>` commit removed :wink:

**Comment by kangax** [comment]
:+1:

**Pull request merged**